buildtest/builders/base.py

```diff
@@ -932,20 +932,20 @@ def _get_variables(self, variables):
 
         return lines
 
-    def _extract_content(self, regex):
-        """Extract content based on the stream and linenum properties
-        from the regex field and return it as a string.
+    def _extract_line(self, regex, content):
```

Comment:
okay so i think this method should just take two arguments `linenum` and `content` that all that it needs. Since this is extracted a line from content we need both parameters. This way we can extend this method in future when extending multiple lines. The `regex` is not needed in this method and we can do this outside the method. 

This method would be typically called like 

```python
linenum = regex.get('linenum')
content = self._extract_line(linenum=linenum, content=input_content)
```

Reply:
The benefit of computing linenum outside the method is we can call this method multiple times if needed like this



```python
linenums=[3, 6, 9]
for i in linenums:
  content += self._extract_line(linenum=i, content=content)  
```

Reply:
👍